src/sentry/models/apitoken.py

```diff
@@ -93,10 +93,10 @@ def organization_id(self) -> int | None:
         try:
             installation = SentryAppInstallation.objects.get_by_api_token(self.id).get()
         except SentryAppInstallation.DoesNotExist:
-            return None
+            installation = None
 
         # TODO(nisanthan): Right now, Internal Integrations can have multiple ApiToken, so we use the join table `SentryAppInstallationToken` to map the one to many relationship. However, for Public Integrations, we can only have 1 ApiToken per installation. So we currently don't use the join table for Public Integrations. We should update to make records in the join table for Public Integrations so that we can have a common abstraction for finding an installation by ApiToken.
-        if installation.sentry_app.status == SentryAppStatus.INTERNAL:
+        if not installation or installation.sentry_app.status == SentryAppStatus.INTERNAL:
```

Comment:
This changes the logic so that if an installation isn't found with the given token ID, it will check for it in SentryAppInstallationToken. This is an unnecessary step if the token is _not_ an internal integration token, but I'm not sure how to tell that here. If anyone knows a better way of doing this please lmk